ExampleCodes/LinearSolvers/GMRES/Poisson/AMReX_GMRES_Poisson.H

```diff
@@ -0,0 +1,193 @@
+#ifndef AMREX_GMRES_MLMG_H_
+#define AMREX_GMRES_MLMG_H_
+#include <AMReX_Config.H>
+
+#include <AMReX_GMRES.H>
+#include <utility>
+
+namespace amrex {
+
+/**
+ * \brief Solve using GMRES with multigrid as preconditioner
+ *
+ * The linear system to solve is provided by MLMG, which is also being used
+ * as the preconditioner.
+ *
+ */
+template <typename MF>
```

Comment:
As a tutorial, it might be better to make this a  non-template concrete class. We can put most of the function definitions in a cpp file.

Reply:
done.